### PR TITLE
valueFormatter setter and getter in tooltip and mini-tooltip

### PR DESCRIPTION
**Pull request by adrm**
## Description

Setter and getter for valueFormatter added to the tooltip and mini-tooltip charts.

## Motivation and Context

Fixes https://github.com/eventbrite/britecharts/issues/594

Without this patch, britecharts only allows you to format values with d3-format. In my case, I am using Angular and I would like to use the Angular number pipe to format every number, also those found inside britechart tooltips.

## How Has This Been Tested?

First, I tried to follow the contribution and style guides as close as possible, trying to mimic already existing code in case of doubt. Then, I tried to touch the least amount of code possible, and do so in the most simple way. Lastly, I ran the test suite with my new tests for the added features, and everything passed with no problem.

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/mini-tooltip.js [hunk 1]
```diff
@@ -378,6 +378,22 @@ define(function(require){
...
         }
 
+        /**
+         * Gets or Sets the formatter function for the value displayed on the tooltip.
+         * Setting this property makes the tooltip ignore numberFormat. Set by default to
+         * d3-format formatter with numberFormat.
+         * @param  {Function} _x Desired formatter function
+         * @return {Function | module} Current valueFormatter or Chart module to chain calls
+         * @public
+         */
```
Comment (Golodhros) [inline src/charts/mini-tooltip.js]: Could you add an @example tag showing how would you use this accessor?

File: src/charts/mini-tooltip.js [hunk 2]
```diff
@@ -378,6 +378,22 @@ define(function(require){
...
+        exports.valueFormatter = function(_x) {
+            if (!arguments.length) {
+                return valueFormatter;
+            }
+            valueFormatter = _x;
+            return this;
```
Comment (Golodhros) [inline src/charts/mini-tooltip.js]: Nit: space before return

File: test/specs/mini-tooltip.spec.js [hunk 3]
```diff
@@ -166,5 +166,23 @@ define(['jquery', 'd3', 'mini-tooltip'], function($, d3, tooltip) {
                 expect(newNameLabel).toBe(testNameLabel);
             });
         });
+
+        it('should allow overriding the value formatting function', () => {
+            let expected = '8',
+                actual;
+
+            tooltipChart.valueFormatter(value => value.toString().length.toString());
```
Comment (Golodhros) [inline test/specs/mini-tooltip.spec.js]: This line could work great for that example

File: src/charts/mini-tooltip.js (hunk 2 above)
Comment (adrm) [reply src/charts/mini-tooltip.js]: You mean to add an empty line before? I copied the structure of other getter/setter methods in the source and I haven't seen an empty line there. If you want me to add it please confirm.

File: src/charts/mini-tooltip.js (hunk 1 above)
Comment (adrm) [reply src/charts/mini-tooltip.js]: Example added for tooltip and mini-tooltip

File: test/specs/mini-tooltip.spec.js (hunk 3 above)
Comment (adrm) [reply test/specs/mini-tooltip.spec.js]: I used the line for the `@example` annotation. Please tell me if I should add anything more.

File: src/charts/mini-tooltip.js (hunk 2 above)
Comment (DalerAsrorov) [reply src/charts/mini-tooltip.js]: Hey @adrm - the formatting in this file is wrong. Please do add an empty line before return. If you want you can also change the format in the same file to keep it consistent. Thanks a lot!

Comment (adrm) [reply src/charts/mini-tooltip.js]: Thanks for the explanation. Will do!

Comment (adrm) [reply src/charts/mini-tooltip.js]: fixed

**Pull request merged**